addons/website/data/website_data.xml

```diff
@@ -309,7 +309,7 @@
     </record>
     <record id="feature_module_career" model="website.configurator.feature">
         <field name="name">Career</field>
-        <field name="description">Publish jobs offer and let people apply</field>
+        <field name="description">Publish job offers and let people apply</field>
```

Comment:
When you change or add a new term in stable version, you have to update the .pot file for the translation (https://github.com/odoo/odoo/wiki/Translations#adding-new-terms-when-suggesting-fixes-in-stable-versions). Technically, we don't even even fix typo in stable version but ... 14.3 we should indeed.

Strange I did not see that in my first review... sorry :/